packages/squiggle-lang/__tests__/Distributions/Invariants/Means_test.res

```diff
@@ -0,0 +1,123 @@
+open Jest
+open Expect
+open TestHelpers
+
+let {
+  algebraicAdd,
+  algebraicMultiply,
+  algebraicDivide,
+  algebraicSubtract,
+  algebraicLogarithm,
+  algebraicPower,
+} = module(DistributionOperation.Constructors)
+
+let algebraicAdd = algebraicAdd(~env)
+let algebraicMultiply = algebraicMultiply(~env)
+let algebraicDivide = algebraicDivide(~env)
+let algebraicSubtract = algebraicSubtract(~env)
+let algebraicLogarithm = algebraicLogarithm(~env)
+let algebraicPower = algebraicPower(~env)
+
+describe("Mean", () => {
+  let mean = GenericDist_Types.Constructors.UsingDists.mean
+
+  let runMean: result<DistributionTypes.genericDist, DistributionTypes.error> => float = distR => {
+    switch distR->E.R2.fmap(mean)->E.R2.fmap(run)->E.R2.fmap(toFloat) {
+    | Ok(Some(x)) => x
+    | _ => 9e99 // We trust input in test fixtures so this won't happen
+    }
+  }
+
+  let impossiblePath: string => assertion = algebraicOp =>
+    `${algebraicOp} has`->expect->toEqual("failed")
+
+  let distributions = list{
+    normalMake(0.0, 1e0),
+    betaMake(2e0, 4e0),
+    exponentialMake(1.234e0),
+    uniformMake(7e0, 1e1),
+    // cauchyMake(1e0, 1e0),
+    lognormalMake(1e0, 1e0),
+    triangularMake(1e0, 1e1, 5e1),
+    Ok(floatMake(1e1)),
+  }
+  let combinations = E.L.combinations2(distributions)
+  let zipDistsDists = E.L.zip(distributions, distributions)
+  let digits = -4
+
+  let testOperationMean = (distOp, description, floatOp, dist1', dist2') => {
```

Comment:
A bit of of hardcoded types would make this much more readable, at least for the code review